#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_jruby_bytecode_version.dpatch by Shyamal Prasad <shyamal@member.fsf.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Set bytecode version for ASM generated bytecodes to target javac 
## DP: version during the build process

@DPATCH@

--- jruby-1.4.0/build.xml.orig	2010-01-09 22:49:27.000000000 -0800
+++ jruby-1.4.0/build.xml	2010-01-09 22:47:35.000000000 -0800
@@ -197,6 +197,7 @@
       <!-- uncomment this line when building on a JVM with invokedynamic
       <jvmarg line="-XX:+InvokeDynamic"/>
       -->
+      <sysproperty key="jruby.bytecode.version" value="${javac.version}"/>
       <arg value="src_gen/annotated_classes.txt"/>
       <arg value="${jruby.classes.dir}"/>
     </java>
